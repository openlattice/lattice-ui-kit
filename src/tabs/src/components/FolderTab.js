import React from 'react';

import Tab from '@material-ui/core/Tab';
import styled from 'styled-components';
import { withStyles } from '@material-ui/core/styles';

import { NEUTRAL, PURPLE } from '../../../colors';

// adjust tab content to take size of bold font
const ContentSpan = styled.span`
  &:before {
    display: block;
    content: ${(props) => (props.content ? `"${props.content}"` : '')};
    font-weight: 600;
    height: 0;
    overflow: hidden;
    visibility: hidden;
  }
`;

const FolderTab = withStyles((theme) => ({
  root: {
    textTransform: 'none',
    minWidth: 72,
    fontWeight: theme.typography.fontWeightRegular,
    marginRight: theme.spacing(1),
    backgroundColor: theme.palette.action.hover,
    minHeight: 32,
    borderRadius: '3px 3px 0px 0px',
    color: NEUTRAL.N700,
    opacity: 1,
    '&:hover': {
      color: PURPLE.P300,
    },
    '&$selected': {
      color: PURPLE.P300,
      backgroundColor: PURPLE.P00,
      fontWeight: theme.typography.fontWeightBold,
    },
    '&:focus': {
      color: PURPLE.P300,
    },
  },
  selected: {},
}))((props) => {
  const labelElement = <ContentSpan content={props.label}>{props.label}</ContentSpan>;
  /* eslint-disable-next-line react/jsx-props-no-spreading */
  return (<Tab disableRipple {...props} label={labelElement} />);
});

export default FolderTab;
