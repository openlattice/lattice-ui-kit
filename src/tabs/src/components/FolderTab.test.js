import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import FolderTab from './FolderTab';

describe('FolderTab', () => {

  describe('default', () => {
    test('should match snapshot', () => {
      const wrapper = mount(<FolderTab label="Tab" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
