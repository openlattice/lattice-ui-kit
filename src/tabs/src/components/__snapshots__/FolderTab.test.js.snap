// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderTab default should match snapshot 1`] = `
.c0:before {
  display: block;
  content: "Tab";
  font-weight: 600;
  height: 0;
  overflow: hidden;
  visibility: hidden;
}

<WithStyles(Component)
  label="Tab"
>
  <Component
    classes={
      Object {
        "root": "Component-root-1",
        "selected": "Component-selected-2",
      }
    }
    label="Tab"
  >
    <WithStyles(ForwardRef(Tab))
      classes={
        Object {
          "root": "Component-root-1",
          "selected": "Component-selected-2",
        }
      }
      disableRipple={true}
      label={
        <styled.span
          content="Tab"
        >
          Tab
        </styled.span>
      }
    >
      <ForwardRef(Tab)
        classes={
          Object {
            "disabled": "Mui-disabled",
            "fullWidth": "MuiTab-fullWidth",
            "labelIcon": "MuiTab-labelIcon",
            "root": "MuiTab-root Component-root-1",
            "selected": "Mui-selected Component-selected-2",
            "textColorInherit": "MuiTab-textColorInherit",
            "textColorPrimary": "MuiTab-textColorPrimary",
            "textColorSecondary": "MuiTab-textColorSecondary",
            "wrapped": "MuiTab-wrapped",
            "wrapper": "MuiTab-wrapper",
          }
        }
        disableRipple={true}
        label={
          <styled.span
            content="Tab"
          >
            Tab
          </styled.span>
        }
      >
        <WithStyles(ForwardRef(ButtonBase))
          className="MuiTab-root Component-root-1 MuiTab-textColorInherit"
          disableRipple={true}
          disabled={false}
          focusRipple={true}
          onClick={[Function]}
          onFocus={[Function]}
          role="tab"
          tabIndex={-1}
        >
          <ForwardRef(ButtonBase)
            className="MuiTab-root Component-root-1 MuiTab-textColorInherit"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            disableRipple={true}
            disabled={false}
            focusRipple={true}
            onClick={[Function]}
            onFocus={[Function]}
            role="tab"
            tabIndex={-1}
          >
            <button
              className="MuiButtonBase-root MuiTab-root Component-root-1 MuiTab-textColorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiTab-wrapper"
              >
                <styled.span
                  content="Tab"
                >
                  <span
                    className="c0"
                    content="Tab"
                  >
                    Tab
                  </span>
                </styled.span>
              </span>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(Tab)>
    </WithStyles(ForwardRef(Tab))>
  </Component>
</WithStyles(Component)>
`;
